pkg/fetch/vendir.go

```diff
@@ -327,3 +336,32 @@ func (v *Vendir) configMapBytes(configMapRef vendirconf.DirectoryContentsLocalRe
 
 	return kyaml.Marshal(configMap)
 }
+
+// TODO: Should this be more generic than just image refs? This would require
+// some parsing and standardization on our end
+func (v *Vendir) skipTLS(url string) bool {
+	hostname := url
+	// adapted based on ggcr's ref parsing
+
+	// remove tag if it exists
+	parts := strings.Split(hostname, ":")
+	if len(parts) > 1 && !strings.Contains(parts[len(parts)-1], "/") {
+		hostname = strings.Join(parts[:len(parts)-1], ":")
+	}
+
+	// remove repo portion
+	parts = strings.SplitN(hostname, "/", 2)
+	if len(parts) == 2 && (strings.ContainsRune(parts[0], '.') || strings.ContainsRune(parts[0], ':')) {
+
+		// The first part of the repository is treated as the registry domain
+		// iff it contains a '.' or ':' character, otherwise it is all repository
+		// and the domain defaults to Docker Hub.
+		hostname = strings.Split(hostname, ":")[0]
+	}
+
+	// handle defaulting of empty hostnames
+	if hostname == "" {
+		hostname = registryDefaultHostname
+	}
+	return v.kcConfig.SkipTLS(hostname)
```

Comment:
i would separate this line from this function so that we have a 1-to-1 ggcr function copy.